easybuild/toolchains/linalg/fujitsussl.py

```diff
@@ -0,0 +1,115 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Support for Fujitsu's SSL library, which provides BLAS/LAPACK support.
+
+:author: Miguel Dias Costa (National University of Singapore)
+"""
+import os
+
+
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.toolchain.constants import COMPILER_FLAGS
+from easybuild.tools.toolchain.linalg import LinAlg
+
+
+FUJITSU_SSL_MODULE_NAME = None
+TC_CONSTANT_FUJITSU_SSL = 'FujitsuSSL'
+
+
+class FujitsuSSL(LinAlg):
+    """Support for Fujitsu's SSL library, which provides BLAS/LAPACK support."""
+    # BLAS/LAPACK support
+    # via lang/tcsds module
+    BLAS_MODULE_NAME = ['lang']
+
+    # no need to specify libraries, compiler driver takes care of linking the right libraries
+    BLAS_LIB = ['']
+    BLAS_LIB_MT = ['']
+    BLAS_FAMILY = TC_CONSTANT_FUJITSU_SSL
+
+    LAPACK_MODULE_NAME = None
+    LAPACK_IS_BLAS = True
+    LAPACK_FAMILY = TC_CONSTANT_FUJITSU_SSL
+
+    BLACS_MODULE_NAME = None
+
+    SCALAPACK_MODULE_NAME = BLAS_MODULE_NAME
+    SCALAPACK_LIB = ['']
+    SCALAPACK_LIB_MT = ['']
+    SCALAPACK_FAMILY = TC_CONSTANT_FUJITSU_SSL
+
+    def _get_software_root(self, name):
+        """Get install prefix for specified software name; special treatment for Fujitsu modules."""
+        if name == 'lang':
+            # Fujitsu-provided  module
+            env_var = 'FJSVXTCLANGA'
+            root = os.getenv(env_var)
+            if root is None:
+                raise EasyBuildError("Failed to determine install prefix for %s via $%s", name, env_var)
+            else:
+                self.log.debug("Obtained install prefix for %s via $%s: %s", name, env_var, root)
+        else:
+            root = super(FujitsuSSL, self)._get_software_root(name)
+
+        return root
+
+    def _set_blas_variables(self):
+        """Setting FujitsuSSL specific BLAS related variables"""
+        super(FujitsuSSL, self)._set_blas_variables()
+        if self.options.get('openmp', None):
+            for flags_var, _ in COMPILER_FLAGS:
+                self.variables.nappend(flags_var, ['SSL2BLAMP'])
+        else:
+            for flags_var, _ in COMPILER_FLAGS:
+                self.variables.nappend(flags_var, ['SSL2'])
+
+    def _set_scalapack_variables(self):
+        """Setting FujitsuSSL specific SCALAPACK related variables"""
+        super(FujitsuSSL, self)._set_scalapack_variables()
+        for flags_var, _ in COMPILER_FLAGS:
+            self.variables.nappend(flags_var, ['SCALAPACK'])
```

Comment:
`-SSL2`/`-SSL2BLAMP` and `-SCALAPACK` are only meant to be used when linking and generate a warning on compiling that is harmless but pollutes the logs

easybuild prepends `-L` to variables appended to `LDFLAGS`, so some refactoring seems to be necessary (?) in order to do this properly (is it worth it?)

Reply:
@migueldiascosta If you think it's important enough, please open an issue to follow up on this.